### PR TITLE
Add CompareToReturnValueProcessor (SonarSource rule 2167)

### PR DESCRIPTION
**Pull request by henry-lp**
Processor for rule [Rule 2167](https://rules.sonarsource.com/java/RSPEC-2167)

Transformation
```diff
-  public int compareTo(CompareToReturnValue a) {
-    return Integer.MIN_VALUE; // Noncompliant 
-  }

+  public int compareTo(CompareToReturnValue a) {
+     return -1;
+  }

```

### CODE REVIEW

**Comment by fermadeiral** [comment]
Conflicts here. This is why I told you to open one PR with a new processor per time, but nothing that cannot be solved anyway :)

**Comment by henry-lp** [comment]
ready for review :) .

File: src/main/java/sonarquberepair/processor/spoonbased/CompareToReturnValueProcessor.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import spoon.reflect.code.CtReturn;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.code.CtLiteral;
+import spoon.reflect.visitor.filter.TypeFilter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompareToReturnValueProcessor extends SQRAbstractProcessor<CtMethod<?>> {
+	public final List<CtMethod<?>> emptyMethods = new ArrayList<>();
```
Comment (fermadeiral) [inline src/main/java/sonarquberepair/processor/spoonbased/CompareToReturnValueProcessor.java:14]: This field is never used.

**Comment by henry-lp** [comment]
review-ready again :)

File: docs/HANDLED_RULES.md [hunk 2]
```diff
@@ -154,6 +155,22 @@ Example:
 ```
 
 -----
+
+#### "compareTo" should not return "Integer.MIN_VALUE" ([Sonar Rule 2167](https://rules.sonarsource.com/java/RSPEC-2167))
+Returning Integer.MIN_VALUE can cause errors because the return value of compareTo is sometimes inversed, with the expectation that negative values become positive. However, inversing Integer.MIN_VALUE yields Integer.MIN_VALUE rather than Integer.MAX_VALUE therefore it's better to return -1.
```
[the remark this answers is not in the document]
Comment (fermadeiral) [reply docs/HANDLED_RULES.md]: Henry, you didn't change everything here.

[the remark this answers is not in the document]
Comment (henry-lp) [reply docs/HANDLED_RULES.md]: Thanks a lot :) .

File: docs/HANDLED_RULES.md [hunk 3]
```diff
@@ -155,13 +155,12 @@ Example:
 -----
 
 #### "compareTo" should not return "Integer.MIN_VALUE" ([Sonar Rule 2167](https://rules.sonarsource.com/java/RSPEC-2167))
-Returning `Integer.MIN_VALUE` can cause errors because the return value of `compareTo` is sometimes inversed, with the expectation that negative values become positive. However, inversing `Integer.MIN_VALUE` yields `Integer.MIN_VALUE` rather than `Integer.MAX_VALUE` therefore it's better to `return -1`.
+Returning `Integer.MIN_VALUE` can cause errors because the return value of `compareTo` is sometimes inversed, with the expectation that negative values become positive. However, inversing `Integer.MIN_VALUE` yields `Integer.MIN_VALUE` rather than `Integer.MAX_VALUE`. Any return `Integer.MIN_VALUE` in a `compareTo` method is then replaced by `return -1`.
```
Comment (fermadeiral) [inline docs/HANDLED_RULES.md:158]: return `Integer.MIN_VALUE` -> `return Integer.MIN_VALUE`

**Comment by fermadeiral** [comment]
Thank you, @henry-lp!

**Comment by henry-lp** [comment]
Thanks , Fernanda :) .

**Comment by monperrus** [comment]
Thank you to both and welcome to CompareToReturnValueProcessor!

Looking forward to the first PRs based on it, what's the timeline for a pilot experiment?

**Comment by henry-lp** [comment]
Currently I am working on [this rule](https://rules.sonarsource.com/java/type/Bug/RSPEC-3067) . Perhaps after 1 more rule while waiting for this repo snapshot deployment to sonartype , then I'll start scanning repos for repair .

**Pull request merged**